test/test_nn.py

```diff
@@ -5090,20 +5091,25 @@ def helper(self, mod, size, dtype, mixed_dtype=False, format=torch.channels_last
             self.assertTrue(out.is_contiguous(memory_format=format))
             self.assertTrue(ref_out.is_contiguous())
             self.assertEqual(out, ref_out)
-            self.assertEqual(bn.weight.grad, ref_bn.weight.grad)
+            self.assertEqual(bn.weight.grad, ref_bn.weight.grad, atol=precision, rtol=precision)
```

Comment:
question: in general is it expected that macos needs higher atol/rtol?

Reply:
For this case, the MPS needs higher atol/rtol for float. I am not familiar with the implementation of MPS, and I guess it may be because the accumulation type is not double for float data type on MPS.
Generally, MPS uses same tolerance value for float data type. As for the float16 data type, based on experience, the comparison between MPS and CPU results may exceed the default tolerance.